libtsuba/include/tsuba/ParquetReader.h

```diff
@@ -17,22 +23,67 @@ class KATANA_EXPORT ParquetReader {
     int64_t length;
   };
 
-  /// \returns a Reader that will read a table from storage location optionally
-  /// reading only part of the table defined by \param slice
+  struct ReadOpts {
+    /// if true (default) make sure canonical types are used and table columns
+    /// are not chunked
+    bool make_safe{true};
+
+    /// if provided, slice the resulting table so that it only contains
+    /// Slice.length rows starting from Slice.offset
+    std::optional<Slice> slice{std::nullopt};
+
+    static ReadOpts Defaults() { return ReadOpts{}; }
+  };
+
+  /// build a reader that will read a table from storage location optionally
+  /// reading only part of the table.
+  /// \param opts an opt structure detailing how reads should behave (see
+  ///    the ReadOpts struct definition for details)
   static katana::Result<std::unique_ptr<ParquetReader>> Make(
-      std::optional<Slice> slice = std::nullopt);
+      ReadOpts opts = ReadOpts::Defaults());
 
```

Comment:
Elegant.

Reply:
Stolen pattern from the arrow codebase! I thought it was good too.